### DISCUSSION TITLE
Struggling setting up dependencies

### DISCUSSION THREAD

**Discussion by gsouf** [NVIDIA-NeMo/NeMo#6874] [Q&A] [answered]
Hi,

I'm facing dependency issue with NeMo. I have no experience with python's dependency management so it's very hard for me to figure out how to solve, I'm hopping to find some help here.

I have a python script that fails as soon as I import NeMo, so my python file literally have this single line of code, nothing else:

```python
from nemo.collections.asr.models.msdd_models import NeuralDiarizer
```

I'm getting import errors with "lightning". I tried many time to re-install dependencies from scratch, using different version... and everytime I got a different error message, but everytime the error message is about a failing import with "lightning". 

My last attempt is after installing `nemo_toolkit[asr]==1.17.0`,  and I get the following error message when I run the python script:

```
File ~/xxx/venv/lib/python3.10/site-packages/pytorch_lightning/strategies/ddp.py:41
     34 from lightning_fabric.utilities.distributed import (
     35     _distributed_available,
     36     _get_default_process_group_backend_for_device,
     37     _init_dist_connection,
     38     _sync_ddp_if_available,
     39 )
     40 from lightning_fabric.utilities.distributed import group as _group
---> 41 from lightning_fabric.utilities.imports import _IS_WINDOWS, _TORCH_GREATER_EQUAL_1_11
     42 from lightning_fabric.utilities.optimizer import _optimizers_to_device
     43 from lightning_fabric.utilities.seed import reset_seed

ImportError: cannot import name '_TORCH_GREATER_EQUAL_1_11' from 'lightning_fabric.utilities.imports' (/xxx/venv/lib/python3.10/site-packages/lightning_fabric/utilities/imports.py)
```

I suppose it all has to do with dependency version, but I'm unable to get it right. Any help is welcome. 

Thanks

**Comment by titu1994** [comment, marked as answer]
We do not currently support pytorch lightning 2.0. Could you try the steps in the readme as follows 

Create new conda env with python 3.9. 
Install cython 
Install torch from pytorch website using conda.
Then install  Nemo toolkit as above (preferably 1.18.1 version).

**Reply by gsouf** [reply]
@titu1994 thanks I managed to solve it. I'm not sure what I was doing wrong, but very likely I installed latest version of lightning.
